### PR TITLE
Support L1 recovery for Banana

### PR DESCRIPTION
**Pull request by cffls**
This commit enables l1 recovery for fork>=12. Tested in kurtosis stack in both Validium and rollup mode.

### CODE REVIEW

File: zk/contracts/l1_abi.go [hunk 1]
```diff
@@ -6,14 +6,20 @@ const SequenceBatchesValidiumAbiElderBerry = "[{\"inputs\":[{\"internalType\":\"
...
+const SequenceBatchesValidiumBanana = "165e8a8d"
+const SequenceBatchesBanana = "b910e0f9"
 
 var SequenceBatchesMapping = map[string]string{
 	SequenceBatchesPreEtrog:           SequenceBatchesPreEtrogAbi,
 	SequenceBatchesIdv5_0:             SequenceBatchesAbiv5_0,
 	SequenceBatchesIdv6_6:             SequenceBatchesAbiv6_6,
 	SequenceBatchesValidiumElderBerry: SequenceBatchesValidiumAbiElderBerry,
+	SequenceBatchesValidiumBanana:     SequenceBatchesAbiBanana,
+	SequenceBatchesBanana:             SequenceBatchesAbiBanana,
```
Comment (cffls) [inline zk/contracts/l1_abi.go:24]: Note: the Abi of banana is shared by both validium and rollup.

**Review by hexoscott — APPROVED** [review]

**Comment by hexoscott** [comment]
Hey @cffls, it looks like this has impacted the unwind tests somehow.  I restarted them after the first failure and the same thing happened again.

**Comment by cffls** [comment]
Thanks @hexoscott , I restarted the test again and this time it passed.

**Comment by hexoscott** [comment]
All good in that case!

**Pull request merged**